libdevcore/CommonIO.cpp

```diff
@@ -121,7 +121,10 @@ string dev::contentsString(std::string const& _file)
 
 void dev::writeFile(std::string const& _file, bytesConstRef _data)
 {
-	ofstream(_file, ios::trunc|ios::binary).write((char const*)_data.data(), _data.size());
+	ofstream s(_file, ios::trunc | ios::binary);
+	s.write(reinterpret_cast<char const*>(_data.data()), _data.size());
+	if (!s)
```

Comment:
It is checked internally in `ofstream::write`. Furthermore, `ofstream` functions should not usually throw exceptions, they only set flags which can be checked by using e.g. `!s`.